### PR TITLE
[Hufi-apikey] feat : add apikey to hufi job launcher 

### PR DESCRIPTION
**Pull request by mrhouzlane**
## Description 

Follow the same iteration for adding api-key to JobLauncher and adapt to Campaigns. 

## Summary of changes

- Added endpoint to generate api-key.

### CODE REVIEW

File: packages/apps/hufi/job-launcher/server/src/common/constants/errors.ts [hunk 1]
```diff
@@ -45,7 +45,10 @@ export enum ErrorAuth {
   NotFound = 'Auth not found',
   InvalidEmailOrPassword = 'Invalid email or password',
   RefreshTokenHasExpired = 'Refresh token has expired',
-  UserNotActive = 'User not active'
+  UserNotActive = 'User not active',
+  ApiKeyCouldNotBeCreatedOrUpdated = 'API key could not be created or updated',
+  ApiKeyNotFound = 'API key not found'
+
```
Comment (portuu3) [inline packages/apps/hufi/job-launcher/server/src/common/constants/errors.ts:51]:
```suggestion
  ApiKeyNotFound = 'API key not found',
```

File: packages/apps/hufi/job-launcher/server/src/modules/job/job.controller.ts [hunk 2]
```diff
@@ -23,8 +23,8 @@ import {
   JobCampaignDto,
 } from './job.dto';
 import { JobService } from './job.service';
-import { JobStatusFilter } from '../../common/enums/job';
-import { Public } from '../../common/decorators';
+import { JobRequestType, JobStatusFilter } from '../../common/enums/job';
```
Comment (portuu3) [inline packages/apps/hufi/job-launcher/server/src/modules/job/job.controller.ts:26]:
```suggestion
import { JobStatusFilter } from '../../common/enums/job';
```

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/apps/hufi/job-launcher/server/src/modules/job/job.controller.ts (hunk 2 above)
Comment (mrhouzlane) [reply packages/apps/hufi/job-launcher/server/src/modules/job/job.controller.ts]: done

File: packages/apps/hufi/job-launcher/server/src/modules/job/job.controller.ts [hunk 3]
```diff
@@ -35,7 +35,8 @@ import { Role } from '../../common/enums/role';
 @Controller('/job')
 export class JobController {
   constructor(private readonly jobService: JobService) {}
-
+
```
[the remark this answers is not in the document]
Comment (mrhouzlane) [reply packages/apps/hufi/job-launcher/server/src/modules/job/job.controller.ts]: I don't get this. removed the line .. 
Done ?

**Comment by portuu3** [comment]
Hufi is going to have its own repo

**Pull request closed without merging**